spring-cloud-starter-tencent-polaris-config/src/main/java/com/tencent/cloud/polaris/config/adapter/PolarisPropertySourceAutoRefresher.java

```diff
@@ -97,45 +122,101 @@ private void registerPolarisConfigPublishEvent() {
-									.changedKeys()) {
-								ConfigPropertyChangeInfo configPropertyChangeInfo = configKVFileChangeEvent
-										.getChangeInfo(changedKey);
-
-								LOGGER.info("[SCT Config] changed property = {}",
-										configPropertyChangeInfo);
-
-								switch (configPropertyChangeInfo.getChangeType()) {
-								case MODIFIED:
-								case ADDED:
-									source.put(changedKey,
-											configPropertyChangeInfo.getNewValue());
-									break;
-								case DELETED:
-									source.remove(changedKey);
-									break;
-								}
+					.addChangeListener((ConfigKVFileChangeListener) configKVFileChangeEvent -> {
+						LOGGER.info(
+								"[SCT Config]  received polaris config change event and will refresh spring context."
+										+ "namespace = {}, group = {}, fileName = {}",
+								polarisPropertySource.getNamespace(),
+								polarisPropertySource.getGroup(),
+								polarisPropertySource.getFileName());
+
+						Map<String, Object> source = polarisPropertySource
+								.getSource();
+						for (String changedKey : configKVFileChangeEvent.changedKeys()) {
+
+							// 1. check whether the changed key is relevant
+							Collection<SpringValue> targetValues = springValueRegistry.get(beanFactory, changedKey);
+							if (targetValues == null || targetValues.isEmpty()) {
+								continue;
 							}
 
-							// rebuild beans with @RefreshScope annotation
-							contextRefresher.refresh();
+							// 2. update the value
+							for (SpringValue val : targetValues) {
+								updateSpringValue(val);
+							}
 						}
+
 					});
 		}
 	}
 
+
+	private void updateSpringValue(SpringValue springValue) {
+		try {
+			Object value = resolvePropertyValue(springValue);
+			springValue.update(value);
+
+			LOGGER.info("Auto update polaris changed value successfully, new value: {}, {}", value,
+					springValue);
+		}
+		catch (Throwable ex) {
+			LOGGER.error("Auto update polaris changed value failed, {}", springValue.toString(), ex);
+		}
+	}
+
+
+	/**
+	 * Logic transplanted from DefaultListableBeanFactory.
+	 *
+	 * @see org.springframework.beans.factory.support.DefaultListableBeanFactory#doResolveDependency(org.springframework.beans.factory.config.DependencyDescriptor,
+	 * java.lang.String, java.util.Set, org.springframework.beans.TypeConverter)
+	 */
+	private Object resolvePropertyValue(SpringValue springValue) {
+		// value will never be null, as @Value and @ApolloJsonValue will not allow that
+		Object value = placeholderHelper
+				.resolvePropertyValue(beanFactory, springValue.getBeanName(), springValue.getPlaceholder());
+
+		if (springValue.isJson()) {
+			value = parseJsonValue((String) value, springValue.getTargetType());
+		}
+		else {
+			if (springValue.isField()) {
+				// org.springframework.beans.TypeConverter#convertIfNecessary(java.lang.Object, java.lang.Class, java.lang.reflect.Field) is available from Spring 3.2.0+
+				if (typeConverterHasConvertIfNecessaryWithFieldParameter) {
```

Comment:
Does SCT also need to be compatible with Spring 3.2.x ? 
What is the intent of this code compatibility ?